src/Service/AutoWiring/ResolverService.php

```diff
@@ -45,7 +45,9 @@ public function resolve(string $className): array
 
         $parameters = $constructor->getParameters();
 
-        return array_map([$this, 'resolveParameter'], $parameters);
+        return array_map(function(\ReflectionParameter $parameter) use ($reflClass) {
```

Comment:
That is not needed because you should get the declaring class from:
`$parameter->getDeclaringClass()->getName()`